src/adapter/z-stack/adapter/zStackAdapter.ts

```diff
@@ -1096,20 +1098,19 @@ class ZStackAdapter extends Adapter {
         });
     }
 
-    private waitForAreqZdo(command: string, payload?: ZpiObjectPayload): {start: () => Promise<ZpiObject>; ID: number} {
+    private waitForAreqZdo<T>(command: string, payload?: ZpiObjectPayload): {start: () => Promise<T>; ID: number} {
         const result = this.znp.waitFor(UnpiConstants.Type.AREQ, Subsystem.ZDO, command, payload);
-        const start = (): Promise<ZpiObject> => {
+        const start = (): Promise<T> => {
             const startResult = result.start();
-            return new Promise<ZpiObject>((resolve, reject) => {
+            return new Promise<T>((resolve, reject) => {
                 startResult.promise
                     .then((response) => {
-                        // Even though according to the Z-Stack docs the status is `0` or `1`, the actual code
-                        // shows it sets the `zstack_ZdpStatus` which contains the ZDO status.
-                        const code = response.payload.status;
-                        if (code !== ZdoStatus.SUCCESS) {
-                            reject(new Error(`ZDO error: ${command.replace('Rsp', '')} failed with status '${ZdoStatus[code]}' (${code})`));
-                        } else {
-                            resolve(response);
+                        const cmd: MtCmdZdo = response.command;
+                        try {
+                            const zdoResponse = BuffaloZdo.readResponse(cmd.zdo.cluterId, response.unpiFrame.data.subarray(cmd.zdo.skip)) as T;
```

Comment:
Looks good and simple. Everything typed 😉

_Might want to rebase before going further, seems you added to the already merged PR branch._